packages/react-client/src/hooks/devices/useDeviceApi.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from "react";
+import type { TrackManager } from "../../types/internal";
+import type { Device } from "../../types/public";
+import type { DeviceManager } from "../../DeviceManager";
+import { useDeviceManager } from "../deviceManagers/useDeviceManager";
+
+type DeviceApiDependencies = {
+  trackManager: TrackManager;
+  deviceManager: DeviceManager;
+};
+
+export const useDeviceApi = ({ trackManager, deviceManager }: DeviceApiDependencies): Device => {
+  const { deviceState, status } = useDeviceManager(deviceManager);
+  const { currentTrack } = trackManager;
+
+  const stream = useMemo(
+    () => currentTrack?.stream ?? deviceState.media?.stream ?? null,
+    [currentTrack, deviceState.media],
+  );
```

Comment:
I'm not quite sure, but maybe we could gain something by adding the full path to the deps array

```suggestion
  const stream = useMemo(
    () => currentTrack?.stream ?? deviceState.media?.stream ?? null,
    [currentTrack?.stream, deviceState.media?.stream],
  );
```